src/webui/webapplication.cpp

```diff
@@ -528,6 +548,9 @@ void WebApplication::configure()
         if (m_trustedReverseProxyList.isEmpty())
             m_isReverseProxySupportEnabled = false;
     }
+
+    if (const QString &apiKey = pref->getWebUIApiKey(); !apiKey.isEmpty() && Utils::APIKey::isValid(apiKey))
```

Comment:
IIRC, this prevents API key from being reset. Shouldn't it be like the following?
```suggestion
    if (const QString &apiKey = pref->getWebUIApiKey(); apiKey.isEmpty() || Utils::APIKey::isValid(apiKey))
```